### PR TITLE
feat(cc-cycle): slice 1 — per-CC cycle substrate (resolver + columns + validation)

### PR DESCRIPTION
**Pull request by flamarion**
## Summary

Slice 1 of 4 for the CC billing cycle work locked in PR #373 spec (`specs/2026-05-28-cc-billing-cycle.md`). Backend-only, no UI. Builds the substrate that Slices 2-4 ride on.

## What lands

- **Migration `061_cc_payment_day_columns`** — adds nullable `payment_day` + `payment_day_relative_month` to `accounts`. No DB default on either (per D7 — the resolver owns the default).
- **`services/cc_cycle_service.py`** — pure-function resolver per D8. Returns `CreditCardCycle(period_start, period_end_inclusive, payment_date, source)`. Implements D2 inclusive close day (`<=`), D3 payment-date math with `calendar.monthrange()` clamp, NULL-means-default fallback, `ValueError` on non-CC.
- **Validation extensions in `account_type_change_service.py`** — `validate_create_payment_day` + `validate_payment_day_cascade` mirror the existing `close_day` pattern. Range checks live in Pydantic (`ge=1, le=31` / `ge=0, le=12`); cascade clears on CC→non-CC; columns rejected on non-CC payloads.
- **Router gate** — `touches_type_or_close_day` extended so PUTs touching the new columns route through the atomic validator path.
- **Model + schema updates** for the two columns.

## Tests

- 16 new in `test_cc_cycle_service.py` covering spec edge cases (a)-(e) + boundary + non-CC `ValueError` regression.
- 14 new in `test_accounts_change_type.py` covering the create + cascade matrix.
- Full backend suite: 1499 passed, 1 skipped, 1 unrelated pre-existing failure (Redis race, tracked in memory).
- Migration round-tripped clean on MySQL 8 in isolated `team-cc-slice-1` stack.

## Out of scope

- Slice 2 (configurable payment-day form + endpoint).
- Slice 3 (`cc_cycle_overrides` table + override endpoint + UI + audit).
- Slice 4 (current-month editor shortcut).

## Review history (pre-submit)

Implemented + spec-compliance reviewed + code-quality reviewed in-session per the user's subagent-driven dev workflow. Quality review surfaced 2 Important + 1 Minor finding (unused param, dup import, missing test assertion) — addressed in `ab125be` and re-verified before push.

### CODE REVIEW

File: backend/app/schemas/account.py [hunk 1]
```diff
@@ -55,6 +57,8 @@ class AccountUpdate(BaseModel):
     account_type_id: Optional[int] = None
     is_active: Optional[bool] = None
     close_day: Optional[int] = Field(default=None, ge=1, le=28)
+    payment_day: Optional[int] = Field(default=None, ge=1, le=31)
+    payment_day_relative_month: Optional[int] = Field(default=None, ge=0, le=12)
```
[the remark this answers is not in the document]
Comment (flamarion) [reply backend/app/schemas/account.py:61]: Fixed in 840804c. `close_day` widened to `le=31` in both `AccountCreate` and `AccountUpdate`; resolver's `calendar.monthrange` clamp now reachable through normal writes. Two new router tests (`test_create_cc_with_close_day_31_accepted`, `test_update_cc_with_close_day_31_accepted`) pin the new contract.

File: backend/app/services/cc_cycle_service.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+
+def _clamp_day(year: int, month: int, day: int) -> date:
+    """Return ``date(year, month, min(day, last_day_of_month))``.
+
+    Mirrors the clamp PFV already uses for ``Organization.billing_cycle_day``
+    in ``lib/date_utils.py``. Prevents ValueError on Feb 29/30/31, Apr 31, etc.
+    """
+    last_day = calendar.monthrange(year, month)[1]
+    return date(year, month, min(day, last_day))
```
[the remark this answers is not in the document]
Comment (flamarion) [reply backend/app/services/cc_cycle_service.py:53]: Fixed in 840804c. `_clamp_day` docstring now points at `billing_service.py:130` as the prior-art reference. Did not extract a shared helper — `billing_service` uses an inline try/except around `date(...)` while this resolver uses an explicit `min(payment_day, last_day)` against `monthrange`; the patterns are similar but not identical, and unifying them is overbuilding for Slice 1.

File: backend/alembic/versions/061_cc_payment_day_columns.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Per-CC cycle substrate (spec 2026-05-28-cc-billing-cycle.md, Slice 1 /
+D3 / D7).
+
+Two new columns on ``accounts``:
+
+- ``payment_day TINYINT NULL``          — the day-of-month the payment
+  is due. NULL means "use resolver default" (day 1). User-settable;
+  the resolver clamps 31 → Feb 28/29, Apr 30, etc. at compute time.
+
+- ``payment_day_relative_month TINYINT NULL``  — how many calendar
```
[the remark this answers is not in the document]
Comment (flamarion) [reply backend/alembic/versions/061_cc_payment_day_columns.py]: Fixed in 840804c. Migration docstring corrected to `INTEGER NULL`. Column DDL was already `sa.Integer()` — mirrors the existing `close_day` column from migration 007 and the rest of the codebase; only the docstring was drifting.

File: backend/tests/services/test_cc_cycle_service.py [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    Close day 25, target Feb 10 → close month Feb, payment month Mar.
+    Mar has 31 days → payment_date = Mar 31.
+    """
+    cycle = resolve_cycle(
+        close_day=25,
+        payment_day=31,
+        payment_day_relative_month=1,
+        target_date=date(2026, 2, 10),
+    )
+    assert cycle.payment_date == date(2026, 3, 31)
```
[the remark this answers is not in the document]
Comment (flamarion) [reply backend/tests/services/test_cc_cycle_service.py]: Fixed in 840804c. Bogus test replaced with `test_payment_day_31_clamps_in_february_non_leap` (target_date Jan 10 2026, payment month = Feb 2026, asserts Feb 28) and a sibling `test_payment_day_31_clamps_in_february_leap` (year 2024, asserts Feb 29). Both would fail if the resolver's `calendar.monthrange` clamp regressed.

File: backend/tests/services/test_cc_cycle_service.py [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    # Feb 29 is valid in 2028 (leap). close_day=29 in Feb → Feb 29.
+    cycle = resolve_cycle(
+        close_day=29,
+        payment_day=None,
+        payment_day_relative_month=None,
+        target_date=date(2028, 2, 15),
+    )
+    assert cycle.period_end_inclusive == date(2028, 2, 29)
+    assert cycle.period_start == date(2028, 1, 30)
+    assert cycle.payment_date == date(2028, 3, 1)
```
[the remark this answers is not in the document]
Comment (flamarion) [reply backend/tests/services/test_cc_cycle_service.py:200]: Fixed in 840804c. Docstring corrected — `period_start = Jan 30` (day after previous close of Jan 29), matching the assertion. The assertion was right; only the docstring drifted.

File: backend/app/routers/accounts.py [hunk 6]
```diff
@@ -192,7 +204,10 @@ async def update_account(
     ip = get_client_ip(request)
 
     touches_type_or_close_day = (
-        body.account_type_id is not None or "close_day" in body.model_fields_set
+        body.account_type_id is not None
+        or "close_day" in body.model_fields_set
+        or "payment_day" in body.model_fields_set
+        or "payment_day_relative_month" in body.model_fields_set
     )
```
[the remark this answers is not in the document]
Comment (flamarion) [reply backend/app/routers/accounts.py]: Fixed in 840804c. Renamed `touches_type_or_close_day` → `touches_type_or_cc_columns` (variable + guard + inline comment). Pure rename; the gate's behavior is unchanged.
